### DISCUSSION TITLE
Mermaid: Pan & Zoom controls: reposition in diagram?

### DISCUSSION THREAD

**Discussion by rphair** [community/community#137171] [Other Feature Feedback, Questions, & Ideas]
Is there a way to "hint" a GitHub Mermaid block to put the pan & zoom controls elsewhere: particularly, in a different corner of the diagram?  

We have a diagram topology that "aims" for the bottom right corner where diagram content is stuck behind the "zoom box" and always obscured in the default scale & translation: https://github.com/cardano-foundation/CIPs/wiki/301.-State-tagging#state-flow-diagram

Yet we have a completely clear field in the upper right corner where it would be really nice to "hint" the GitHub Mermaid previewer to put that zoom-box.

> [!NOTE]
> I'm not directing this to the Mermaid community because the zoom-box appears differently on GitHub than in other Mermaid applications (e.g. the official [Mermaid Live Editor](https://mermaid.live)) and it uses the same widget set as other GitHub features: therefore it has to be the particular GitHub integration of Mermaid that is rendering & positioning the zoom-box.

**Comment by jasonwilliams** [comment]
I’ve added a tangentially related issue which asks to hide the controls unless the user interacts with the chart (this is similar to how videos work in the browser).

https://github.com/orgs/community/discussions/178929

**Comment by birdyhash** [comment]
Hi!

What you’re seeing is specific to **GitHub Mermaid integration**. The pan & zoom controls (“zoom box”) on GitHub are **positioned automatically by their renderer**, and unfortunately, there’s **no supported way to hint or configure its placement** via Markdown or Mermaid directives.

### Key points

* **GitHub-specific behavior**:

  * The zoom/pan widget is part of GitHub’s UI, not the standard Mermaid library.
  * Other editors (like Mermaid Live Editor) allow custom positioning, but GitHub strips any configuration for security and consistency.

* **No “hint” option**:

  * Inline CSS, HTML, or Mermaid directives cannot change where the zoom controls appear.

### Workarounds

1. **Reposition your diagram content**

   * Adjust your nodes/layout so the bottom-right area is free; move important content away from where the zoom box appears.

2. **Split the diagram**

   * Break complex diagrams into multiple Mermaid blocks to avoid overlap.

3. **Use an image**

   * Render the diagram externally (`mmdc` CLI, Mermaid Live Editor) and embed as an image.
   * You lose interactive pan & zoom but gain full control of layout.

4. **Feature request**

   * You could submit a GitHub Feedback request asking for configurable zoom/pan control placement. That’s the only way it might eventually be supported.

💡 **Practical tip for your diagram**

* For your Cardano CIP state-flow diagram, try temporarily flipping or shifting the graph layout so critical nodes avoid the bottom-right corner. This keeps everything visible without waiting for a feature update.

## Example

```
flowchart LR
    %% Invisible spacer to push content away from bottom-right
    subgraph spacer[ ]
        direction LR
        A1([ ]) --> A2([ ])
    end

    %% Main diagram content
    subgraph state_flow["State Flow"]
        direction TB
        Start --> State1
        State1 --> State2
        State2 --> End
    end

    %% Connect spacer to main diagram to occupy the bottom-right
    spacer --> End
```

---

<div align="center">

![repos](https://lastrepos-vert.vercel.app//api/badge?user=ogcae\&style=compact\&theme=github_dark\&icon=emoji\&limit=2)
[![ogcae](https://img.shields.io/badge/-ogcae-black?style=for-the-badge\&logo=github\&logoColor=white)](https://github.com/ogcae)

</div>

**Comment by alessandrobologna** [comment]
I couldn't get the suggested solution to work, but this worked:

```mermaid
%%{init: {"flowchart": {"diagramPadding": 150}}}%%
flowchart LR
    A1["ctx.step('a', ...)"] --> A2["ctx.step('b', ...)"]
    A2 --> A3["ctx.wait(60s)"]
    A3 --> A4(("SUSPENDS"))
    A4 --> A5["Lambda frozen"]
```

instead of just

```mermaid
flowchart LR
    A1["ctx.step('a', ...)"] --> A2["ctx.step('b', ...)"]
    A2 --> A3["ctx.wait(60s)"]
    A3 --> A4(("SUSPENDS"))
    A4 --> A5["Lambda frozen"]
```

Basically it adds a padding that pushes the controls to the bottom:

```code
%%{init: {"flowchart": {"diagramPadding": 150}}}%%
flowchart LR
    A1["ctx.step('a', ...)"] --> A2["ctx.step('b', ...)"]
    A2 --> A3["ctx.wait(60s)"]
    A3 --> A4(("SUSPENDS"))
    A4 --> A5["Lambda frozen"]
```
